### PR TITLE
[PUB-2567] Add campaign tag button to sent posts

### PR DESCRIPTION
**Pull request by chayjay**
## Description

Adds campaign tag button to sent post tab and adds a few fixes to the campaign tag button:
-  The button will now be the same color as the campaign color on hover
- The button will direct you to the url campaigns/${campaign_id}/scheduled
-  Text color on button will update based on campaign color

## Context & Notes
PUB-2567

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [x] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/shared-components/PostList/index.jsx [hunk 1]
```diff
@@ -54,15 +54,18 @@ const renderPost = ({
...
   onSwapPosts,
   isSent,
   isBusinessAccount,
   isPastReminder,
   hasFirstCommentFlip,
+  hasCampaignsFeature,
   userData,
   onPreviewClick,
 }) => {
+  const campaignId = post.campaignDetails ? post.campaignDetails.id : null;
```
Comment (MayaUribe) [inline packages/shared-components/PostList/index.jsx]:
```suggestion
  const campaignId = post.campaignDetails?.id ?? null;
```

Comment (MayaUribe) [reply packages/shared-components/PostList/index.jsx]: @chayjay maybe we could use our new optional chaining here 😄

**Comment by MayaUribe** [comment]
@chayjay you probably already noticed, but you probably need to update the snapshots file for that travis-ci check to pass ☺️ mentioning it just in case.

File: packages/shared-components/PostList/index.jsx (hunk 1 above)
Comment (chayjay) [reply packages/shared-components/PostList/index.jsx]: Great idea!

**Comment by chayjay** [comment]
> @chayjay you probably already noticed, but you probably need to update the snapshots file for that travis-ci check to pass ☺️ mentioning it just in case.

Thank you for the reminder! Should be fixed now 😄

**Pull request merged**